### ISSUE TITLE
Correction fails for large number of pixels

### ISSUE THREAD

**Issue by sk1p** [LiberTEM/LiberTEM#889]
Calling `libertem.corrections.detector.correct` with a large number of dead pixel coordinates fails with the following message:

```
UnsupportedError: Failed in nopython mode pipeline (step: ensure IR is legal prior to lowering)
Tuple 'repair_environments' length must be smaller than 1000.
Large tuples lead to the generation of a prohibitively large LLVM IR which causes excessive memory pressure and large compile times.
As an alternative, the use of a 'list' is recommended in place of a 'tuple' as lists do not suffer from this problem.

File "../source/LiberTEM/src/libertem/corrections/detector.py", line 9:
@numba.njit(cache=True)
def _correct_numba_inplace(buffer, dark_image, gain_map, exclude_pixels, repair_environments):
^
```

This was with slightly above 1k to 5k (correction: 12k) nnz entries in the dead pixel mask (amounts to less than 1% of all pixels, which is a reasonable number).

This happens with `numba==0.51.2` and `numba==0.50`. I think this error message may actually hint at a problem in the code: tuple length is part of the type when compiling, and changing to a different dead-pixel list forces a re-compile. This could be nice if the amount of pixels to be corrected is small (I guess some unrolling could be happening?), but for larger correction sets, this fails spectacularly.

When calling `correct` with something like this:

```python
img2 = correct(buffer=img.reshape((1,) + img.shape), excluded_pixels=coords.coords[:, :1000], sig_shape=tuple(img.shape))
```

(Here `coords` is a `sparse.COO` array) This works around the problem of not compiling, but compilation is very very slow (takes 1 minute and more!!). Applying this works in reasonable time (on the order of 200ms).

**Comment by sk1p** [comment]
A possible fix could be to switch to a `numba` [typed list](https://numba.pydata.org/numba-doc/latest/reference/pysupported.html#typed-list), or we could use a different (dense) encoding for the ragged coordinate array.

**Comment by sk1p** [comment]
First run:

```
CPU times: user 1min 47s, sys: 1.06 s, total: 1min 49s
Wall time: 1min 48s
```

Second run:

```
CPU times: user 144 ms, sys: 0 ns, total: 144 ms
Wall time: 143 ms
```

**Comment by sk1p** [comment]
Fixed by #890
